fs_storage/models/fs_storage.py

```diff
@@ -339,6 +339,17 @@ def _get_fs_options(self):
             return options
         return self._eval_options_from_env(self.json_options)
 
+    def _get_marker_file_name(self):
+        return ".odoo_fs_storage_%s.marker" % self.id
+
+    def _check_connection(self, fs=False):
+        if not fs:
+            fs = self.fs
+        marker_file_name = self._get_marker_file_name()
+        if not fs.ls(marker_file_name, detail=False):
```

Comment:
@JordiMForgeFlow @simahawk even by installing `fsspec@master` I get the error `FileNotFoundError: [Errno 2] No such file` on this line.

FYI it has been tested with `sftp` protocol, the issue is raised on this line exactly: https://github.com/fsspec/filesystem_spec/blob/master/fsspec/implementations/sftp.py#L127

Should we do this check inside a `try/except` instead? I even tried `self.fs.info(path)` instead of `ls` but it is raising the same `FileNotFound` exception.

Reply:
Why don't we use the 'exists' method?

Reply:
@lmignon the problem I see with 'exists' is that it returns either True or False, it covers other potential errors that could raise from the connection. I think that the 'ls' with a try/except for the 'filenotfounderror' should work. WDYT?

Reply:
:+1:

Reply:
Should we also check for `PermissionError`? E.g. with a forbidden folder:
`PermissionError: [Errno 13] Permission denied: '/root/.odoo_fs_storage_4.marker'`

Reply:
@sebalix how would you then handle this error? We would not be able to use the marker file, right?

Reply:
Maybe we should not handle it you are right, and let it pop-up and get an error about it.

Reply:
@sebalix I have added the handling for the FileNotFoundError. I have tested locally and it works fine. Could you check from your side?

Reply:
@JordiMForgeFlow now it works as expected, thanks!